packages/std/src/imports.rs

```diff
@@ -75,6 +111,43 @@ impl Storage for ExternalStorage {
             write_db(key_ptr, value_ptr);
         }
     }
+
+    fn remove(&mut self, key: &[u8]) {
+        // keep the boxes in scope, so we free it at the end (don't cast to pointers same line as build_region)
+        let key = build_region(key);
+        let key_ptr = &*key as *const Region as *const c_void;
+        unsafe {
+            remove_db(key_ptr);
+        }
+    }
+}
+
+#[cfg(feature = "iterator")]
+/// ExternalIterator makes a call out to next
+/// We only allow one open iterator at a time, so no need to pass references
+/// it automatically refers to result of last range call
```

Comment:
Do we want to keep this limitation? Couldn't scan return any kind of iterator identifier?

Reply:
I had this before and removed it as I got segfaults.

It's actually tricky enough to handle one `Option<Iterator>`, storing `HashMap<u32, Iterator>` is more tricky. I would add a follow-up issue to enhance this. For now, this should cover most use cases, and we can extend it (before 0.8)

Reply:
Created #190